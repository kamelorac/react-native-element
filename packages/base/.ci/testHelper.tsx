import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
