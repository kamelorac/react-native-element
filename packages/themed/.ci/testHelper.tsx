import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
