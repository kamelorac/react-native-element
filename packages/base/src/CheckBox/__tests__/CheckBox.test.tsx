import React from 'react';
import { CheckBox } from '..';
import { renderWithWrapper } from '../../../.ci/testHelper';
import { Pressable, View, Text, Image } from 'react-native';

describe('CheckBox Component', () => {
  it('should match snapshot', () => {
    const component = renderWithWrapper(<CheckBox checked />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should use Pressable as default component', () => {
    const { wrapper } = renderWithWrapper(<CheckBox checked />);
    expect(wrapper.findAllByType(Pressable).length).toBe(1);
  });

  it('should allow to pass custom component', () => {
    const { wrapper } = renderWithWrapper(
      <CheckBox checked Component={View} />
    );
    expect(wrapper.findByType(View)).toBeTruthy();
  });

  it('should render title in Text', () => {
    const component = renderWithWrapper(
      <CheckBox title="Custom Text" checked checkedTitle="Custom Text" />
    );
    expect(component.queryByText('Custom Text')).toBeTruthy();
  });

  it('should render with wrapperStyle', () => {
    const { wrapper, queryByText } = renderWithWrapper(
      <CheckBox
        title="Custom Text"
        checked
        wrapperStyle={{ backgroundColor: 'red' }}
      />,
      'RNE__CheckBox__Wrapper'
    );
    expect(queryByText('Custom Text')).toBeTruthy();
    expect(wrapper.findByType(View).props.style).toMatchObject({
      backgroundColor: 'red',
    });
  });

  it('should render with textStyle and fontFamily', () => {
    const { wrapper } = renderWithWrapper(
      <CheckBox
        title="Custom Text"
        checked
        textStyle={{ color: 'red' }}
        fontFamily="serif"
      />,
      'RNE__CheckBox__Title'
    );
    expect(wrapper.props.style).toMatchObject({
      color: 'red',
      fontFamily: 'serif',
    });
  });

  it('should allow title to be custom component', () => {
    const CustomText = 'Custom Component!';
    const { queryByText, queryByTestId } = renderWithWrapper(
      <CheckBox
        checked
        title={
          <View>
            <Text testID="custom-text">{CustomText}</Text>
          </View>
        }
      />
    );
    expect(queryByText(CustomText)).toBeTruthy();
    expect(queryByTestId('custom-text')).toBeTruthy();
  });

  it('should render with icon and checked', () => {
    const { wrapper } = renderWithWrapper(
      <CheckBox checked iconType="font-awesome" checkedColor="red" />,
      'RNE__Checkbox__Icon'
    );
    expect(wrapper.props.style[2]).toMatchObject({
      fontFamily: 'FontAwesome',
      fontWeight: 'normal',
      fontStyle: 'normal',
    });
  });

  it('should allow custom checked Icon', () => {
    const { wrapper } = renderWithWrapper(
      <CheckBox
        checked
        checkedIcon={
          <Image
            source={{ uri: 'https://image.ibb.co/jcY95H/checked.png' }}
            style={{ width: 30, height: 30 }}
          />
        }
      />
    );
    expect(wrapper.findByType(Image).props.source).toMatchObject({
      uri: 'https://image.ibb.co/jcY95H/checked.png',
    });
  });

  it('should allow custom unchecked Icon', () => {
    const { wrapper } = renderWithWrapper(
      <CheckBox
        checked={false}
        uncheckedIcon={
          <Image
            source={{ uri: 'https://image.ibb.co/fda0Cx/no_check.png' }}
            style={{ width: 30, height: 30 }}
          />
        }
      />
    );
    expect(wrapper.findByType(Image).props.source).toMatchObject({
      uri: 'https://image.ibb.co/fda0Cx/no_check.png',
    });
  });

  it('should allow passing props to the title', () => {
    const { wrapper } = renderWithWrapper(
      <CheckBox checked title="Demo Text" titleProps={{ numberOfLines: 2 }} />,
      'RNE__CheckBox__Title'
    );
    expect(wrapper.props.children).toBe('Demo Text');
    expect(wrapper.props.numberOfLines).toBe(2);
  });
});
