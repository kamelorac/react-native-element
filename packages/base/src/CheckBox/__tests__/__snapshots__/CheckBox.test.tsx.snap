// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      {
        "checked": true,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#ffffff",
        "margin": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "padding": 10,
      }
    }
    testID="RNE__CheckBox__Wrapper"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          [
            {
              "color": "#2089dc",
              "fontSize": 24,
            },
            {
              "minWidth": 24,
            },
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="RNE__Checkbox__Icon"
      >
        
      </Text>
    </View>
  </View>
</View>
`;
